packages/terminal/src/browser/base/terminal-widget.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { Widget } from '@phosphor/widgets';
+import { Event } from '@theia/core';
+
+/**
+ * Terminal UI widget.
+ */
+export const TerminalWidget = Symbol('TerminalWidget');
+export interface TerminalWidget extends Widget {
+    /**
+     * Start terminal and return terminal id.
+     * @param id - terminal id.
+     */
+    start(id?: number): Promise<number>;
+
+    /**
+     * Send text to the terminal server.
+     * @param text - text content.
+     */
+    sendText(text: string): void;
+
+    /**
+     * Event which fires when terminal did closed. Event value contains closed terminal widget definition.
+     */
+    onTerminalDidClose: Event<TerminalWidget>;
+}
+
+export function isTerminalWidget(widget: Widget | undefined): widget is TerminalWidget {
+    if (!widget) {
+        return false;
+    }
+
+    const widgetToCheck = <TerminalWidget>widget;
+    if (widgetToCheck.start && widgetToCheck.sendText && widgetToCheck.onTerminalDidClose) {
```

Comment:
for this case is fine, but generally it is better to use `in` operator to check property existence, since a property can be undefined, i.e.
```ts
return !!widget && ('start' in widget) && ('sendText' in widget) && ('onTerminalDidClose' in widget);